src/main/java/se/liu/ida/hefquin/engine/wrappers/graphqlwrapper/utils/SPARQL2GraphQLHelper.java

```diff
@@ -0,0 +1,247 @@
+     */
+    public static String determineSgpType(final Set<TP> sgp, final GraphQL2RDFConfiguration config){
+        for(final TP t : sgp){
+            final Node predicate = t.getTriplePattern().asJenaTriple().getPredicate();
+            final Node object = t.getTriplePattern().asJenaTriple().getObject();
+
+            if(predicate.isURI() && predicate.getURI().startsWith(config.getPropertyPrefix())){
+                return config.getClassFromPropertyURI(predicate.getURI());
+            }
+            else if(predicate.isURI() && predicate.getURI().equals(config.getClassMembershipURI())){
+                if(object.isURI() && object.getURI().startsWith(config.getClassPrefix())){
+                    final int splitIndex = config.getClassPrefix().length();
+                    return object.getURI().substring(splitIndex);
+                }
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * @return a map consisting of what can be used as arguments from the given @param sgp
+     * The predicate from a TP needs to be a property URI and the object needs to be a literal
+     */
+    public static final Map<String,LiteralLabel> getSgpArguments(final Set<TP> sgp, final GraphQL2RDFConfiguration config){
+        final Map<String,LiteralLabel> args = new HashMap<>();
+        for(final TP t : sgp){
+            final Node predicate = t.getTriplePattern().asJenaTriple().getPredicate();
+            final Node object = t.getTriplePattern().asJenaTriple().getObject();
+
+            if(predicate.isURI() && predicate.getURI().startsWith(config.getPropertyPrefix())){
+                if(object.isLiteral() && object.getLiteral().isWellFormed()){
+                    // Remove prefix and suffix before adding to map
+                    final String s = config.removePropertyPrefix(predicate.toString());
+                    final String argName = config.removePropertySuffix(s);
+                    args.put(argName,object.getLiteral());
+                }
+            }
+        }
+        return args;
+    }
+
+    /**
+     * Checks whether a set of SGP arguments, @param sgpArguments , meets the necessery arguments for
+     * an endpoint @param entrypointArguments . If @param checkAll is true then all arguments from the entrypoint
+     * must exist in the SGP. Otherwise atleast one matching argument is enough as long as @param sgpArguments
+     * isn't an empty set.
+     */
+    public static boolean hasNecessaryArguments(final Set<String> sgpArguments, final Set<String> entrypointArguments, 
+            final boolean checkAll){
+        
+        if(sgpArguments.isEmpty()){
+            return false;
+        }
+
+        if(checkAll){
+            return sgpArguments.containsAll(entrypointArguments);
+        }
+        else{
+            for(final String argName : sgpArguments){
+                if(entrypointArguments.contains(argName)){
+                    return true;
+                }
+            }
+        }
```

Comment:
these are essentially two different methods, please just split this instead of providing a boolean flag.